src/service/url-expander/expander.js

```diff
@@ -140,11 +140,12 @@ export class Expander {
               // precedence of resolution choices in #expandBinding_ later.
               name: match.name,
               prioritized: this.bindings_[match.name],
+              encode,
             };
           } else {
             // or the global source
-            binding = this.variableSource_.get(match.name);
-            binding.name = match.name;
+            binding = Object.assign({}, this.variableSource_.get(match.name),
```

Comment:
:-( don't quite like this.
can we just pass `encode` to evaluateBinding?

Reply:
Its tough because several bindings may be pushed into the stack before `evaluateBinding` is called. Also the bindings not from the `variableSource` have to be constructed this way a few lines above.